drones/image_processing/normalization.py

```diff
@@ -0,0 +1,250 @@
+    img_b = cdf_final_b[b]
+    img_g = cdf_final_g[g]
+    img_r = cdf_final_r[r]
+
+    img_out = cv.merge((img_b, img_g, img_r))
+    return img_out
+
+
+def image_centralization(image: np.ndarray) -> np.ndarray:
+    """Centralize image value. It should be used after apply zero_one norm
+
+    Parameters:
+    ----------
+    image : np.ndarray
+        image to be centered
+
+    Returns:
+    ----------
+    image : np.ndarray
+        centered version of input image
+    """
+    return (image - image.mean()) * 2
+
+
+def clahe_gray_norm(image: np.ndarray) -> np.ndarray:
+    """Apply clahe method on gray image
+
+    Parameters:
+    ----------
+    image : np.ndarray
+        image to be normalized
+
+    Returns:
+    ----------
+    image : np.ndarray
+        normalized version of input image
+    """
+    clahe = cv.createCLAHE(clipLimit=2.0, tileGridSize=(8, 8))
+    return clahe.apply(image)
+
+
+def clahe_color_norm(image: np.ndarray) -> np.ndarray:
+    """Apply clahe method on color image
+
+    Parameters:
+    ----------
+    image : np.ndarray
+        image to be normalized
+
+    Returns:
+    ----------
+    image : np.ndarray
+        normalized version of input image
+    """
+    lab = cv.cvtColor(image, cv.COLOR_BGR2LAB)
+
+    lab_planes = cv.split(lab)
+
+    clahe = cv.createCLAHE(clipLimit=2.0, tileGridSize=(8, 8))
+
+    lab_planes[0] = clahe.apply(lab_planes[0])
+
+    lab = cv.merge(lab_planes)
+
+    return cv.cvtColor(lab, cv.COLOR_LAB2BGR)
+
+
+def normalization(
+    image: np.ndarray,
+    minmax: bool = False,
+    zero_one: bool = False,
+    gray: bool = False,
+    histogram_equalization: bool = False,
+    centralization: bool = False,
+    clahe: bool = False,
+) -> np.ndarray:
+    """Apply chosen norms on given image. Default all norms are switch off
+
+    Parameters:
+    ----------
+    image : np.ndarray
+        image to be normalized
+    minmax: bool
+        apply minmax norm on image
+    zero_one: bool
+        apply zero_one norm on image
+    gray: bool
+        make image grayscale
+    histogram_equalization: bool
+        equalize histogram of all channels of image, depending on which scale image is
+    centralization: bool
+        make image central
+    clahe:bool
+        apply clahe method on image
+
+    Returns:
+    ----------
+    image : np.ndarray
+        normalized version of input image
+    """
+    function_dict = {
+        min_max_norm: minmax,
+        zero_one_norm: zero_one,
+        gray_norm: gray,
+        image_centralization: centralization,
+    }
+    if gray:
+        function_dict[histogram_equalization_gray] = histogram_equalization
+        function_dict[clahe_gray_norm] = clahe
+    else:
+        function_dict[histogram_equalization_color] = histogram_equalization
+        function_dict[histogram_equalization_luminance] = histogram_equalization
+        function_dict[clahe_color_norm] = clahe
+
+    for function, execute in function_dict.items():
```

Comment:
It would be better to rename these variables so that it is easier to understand their purpose e.g. `norm_function`, `execute_norm`, `norm_function_dict` or something similar.